### PR TITLE
qa: wait rank 0 to become up:active state before mounting fuse client

### PR DESCRIPTION
**Pull request by lxbsz**
When setting the ec pool to the layout the filesystem may not be
ready, so when mounting a fuse client it will fail. To fix this we
need to wait at least the rank 0 to be in up:active state.

Fixes: https://tracker.ceph.com/issues/55824
Signed-off-by: Xiubo Li <xiubli@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: qa/tasks/cephfs/filesystem.py [hunk 1]
```diff
@@ -713,6 +713,11 @@ def create(self):
     def run_client_payload(self, cmd):
         # avoid circular dep by importing here:
         from tasks.cephfs.fuse_mount import FuseMount
+
+        # Wait for mds.0 to reach up::active state before mounting
+        # cephfs client in run_client_payload()
+        self.wait_for_state('up:active', timeout=30, rank=0)
+
```
Comment (joscollin) [inline qa/tasks/cephfs/filesystem.py:720]: a timeout would still mount?

Comment (lxbsz) [reply qa/tasks/cephfs/filesystem.py:720]: If after `30` seconds the rank 0 still couldn't reach `up:active` state, the `wait_for_state()` will `raise RuntimeError`.

**Review by kotreshhr — APPROVED** [review]

**Review by joscollin — APPROVED** [review]

**Review by neesingh-rh — APPROVED** [review]

**Review by dparmar18 — APPROVED** [review]

File: qa/tasks/cephfs/filesystem.py [hunk 2]
```diff
@@ -713,6 +713,11 @@ def create(self):
     def run_client_payload(self, cmd):
         # avoid circular dep by importing here:
         from tasks.cephfs.fuse_mount import FuseMount
+
+        # Wait for mds.0 to reach up::active state before mounting
+        # cephfs client in run_client_payload()
+        self.wait_for_state('up:active', timeout=30, rank=0)
```
Comment (batrick) [inline qa/tasks/cephfs/filesystem.py:719]:
```suggestion
        self.wait_for_daemons()
```

there may be more than one active MDS when this method is called in future places.

Comment (dparmar18) [reply qa/tasks/cephfs/filesystem.py:719]: default timeout for `wait_for_daemons()` is 120 seconds, so `wait_for_daemons(timeout=30)` should suffice

Comment (lxbsz) [reply qa/tasks/cephfs/filesystem.py]: Done, and will use the default `120 seconds`.

Comment (lxbsz) [reply qa/tasks/cephfs/filesystem.py]: Thanks @batrick @dparmar18

**Review by vshankar — APPROVED** [review]

**Review by batrick — APPROVED** [review]

**Comment by vshankar** [comment]
https://tracker.ceph.com/projects/cephfs/wiki/Master#2022-June-20

**Pull request merged**